### PR TITLE
Fix #3979 marketplace shipping

### LINKED ISSUE #3979: Receive server error when shipping order in Marketplace
## Issue Description
Receive an error in the server console when shipping an order in the marketplace. Notification does not appear to be sent and order is not marked "shipped"

```
Exception while invoking method 'orders/shipmentShipped' WriteError({"code":11000,"index":0,"errmsg":"E11000 duplicate key error collection: meteor.OrderSearch index: _id_ dup key: { : \"hCWhcEw3KhbaaQ3Et\" }","op":{"_id":"hCWhcEw3KhbaaQ3Et","shopId":"J8Bhq3uTtdgwZx3rz","shippingName":"Brent Hoover","shippingPhone":"3236873265","billingName":"Brent Hoover","userEmails":["brent@reactioncommerce.com"],"shippingAddress":{"address":"2110 Main St. Suite 207","postal":"90405","city":"Santa Monica","region":"CA","country":"US"},"billingAddress":{"address":"2110 Main St. Suite 207","postal":"90405","city":"Santa Monica","region":"CA","country":"US"},"shippingStatus":"New","billingStatus":"completed","orderTotal":18.99,"orderDate":"2018/03/12","billingPhone":"3236873265","billingCard":"Visa 4242","currentWorkflowStatus":"coreOrderWorkflow/processing","product":{"title":["Dumb Product","Reaction Bag of Beans"]},"variants":{"title":["Size Nine","One pound bag"],"optionTitle":["Size 9","One pound bag"]}}})
```

## Steps to Reproduce

1. Configure for marketplace and second shop
1. Create product in second shop
1. In Primary shop place an order with a product from the main shop and from the second shop
1. In the dashboard approve, capture, and attempt to ship the order
1. Observe the error and that the order is not marked shipped 

### Versions
```
Node: 8.9.4
NPM: 5.6.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.28.0
Reaction: 1.9.0
Reaction branch: release-1.9.0
Docker: 17.12.0-ce
```

### PR DESCRIPTION
**Pull request by Akarshit**
Resolves #3979 
Impact: **critical**  
Type: **bugfix**

## Issue
This issue is a timing issue. When a order is updated, a [hook](https://github.com/reactioncommerce/reaction/blob/release-1.9.0/imports/plugins/included/search-mongo/server/hooks/search.js#L51) is run to remove the old `OrderSearch` doc and add the new one. Since the orders is displayed twice, it is updated twice(and the hook runs almost together). 
In general the order of execution is
```
remove doc
insert doc
remove doc
insert doc
```
but in the rare case it becomes
```
remove doc
remove doc
insert doc
insert doc
```
You get the error in the issue. 

One other way to reproduce the issue(and with much more success) is to keep on clicking on the `Approve` or `Shipped` button(they aren't disabled after first click).

## Solution
There is only one place in the code where the record is inserted in the `OrderSearch` collection. At that place I changed the `insert` operation to a `upsert` operation, which will prevent the duplicate id error.

## Testing
1. Configure for marketplace and second shop
1. Create product in second shop
1. In Primary shop place an order with a product from the main shop and from the second shop
1. In the dashboard approve, capture, and attempt to ship the order
1. Observe no error and that the order is marked shipped

OR

1. Place order for a single product.
2. Fo to the dashboard and keep clicking on the `Approve` and then on the `Shipped` button.
1. Observe no error in the console.

### CODE REVIEW

File: imports/plugins/included/search-mongo/server/methods/searchcollections.js [hunk 1]
```diff
@@ -242,7 +242,7 @@ export function buildOrderSearchRecord(orderId) {
   orderSearch.variants.title = order.items.map((item) => item.variants && item.variants.title);
   orderSearch.variants.optionTitle = order.items.map((item) => item.variants && item.variants.optionTitle);
 
-  OrderSearch.insert(orderSearch);
+  OrderSearch.upsert(orderId, { $set: { ...orderSearch } });
```
Comment (zenweasel) [inline imports/plugins/included/search-mongo/server/methods/searchcollections.js]: I think additionally we should surround this in a `try/catch`. There's no reason that anything that happens here should block settings the shipping status or anything else. So we should just catch and log any errors.

Comment (Akarshit) [reply imports/plugins/included/search-mongo/server/methods/searchcollections.js]: OK doing that.

**Comment by Akarshit** [comment]
@zenweasel Updated.

**Comment by zenweasel** [comment]
@Akarshit You have failing tests here

**Comment by Akarshit** [comment]
@zenweasel `meteor npm install` is not working in cicleci. I am pretty sure I didn't break that.
Maybe I can try rebuilding this again with a fake commit.

**Comment by Akarshit** [comment]
@zenweasel Cleared the tests.

**Pull request merged**